### DISCUSSION TITLE
Roadmap, v1.0 & use cases

### DISCUSSION THREAD

**Discussion by samuelsainz** [heyjaywilson/peakfit#60] [General]
Hey team! 

I'm planning to help building this app as much as I can and I was thinking that it would be great to know a bit more about the vision of the app. I’ve been able to figure out quite a bit by exploring the repo and reviewing the open issues, but having some docs could help to not making bad assumptions. 

Here is some of the info that I think it would be very helpful:

* What is the main goal of the app? 
* Who is the app intended for? 
* What are the main features? 

To support all of this it may help to have some sort of Roadmap with the planned features for the first version of the app—and even for future versions. Also, some use cases or stories to describe the features would be awesome, that way we all could help writing new Issues and proposing new ideas. 

@heyjaywilson I know you probably already have a lot of this things. I don’t want to be too forward or intrusive, especially since I'm not entirely sure how things are typically done in the open-source community—I haven't had the opportunity to collaborate in building an app from scratch in an open-source repository before—so please feel free to guide me if I'm stepping outside any usual boundaries

**Comment by heyjaywilson** [comment]
I love the enthusiasm!!!

This is a lowkey project right now that I’m trying to get some basic features in to solve a personal want that I had to track my strength exercises. It sounds like it really resonates with you also, which means it’s probably wanted by other people. 

Like I said in another discussion, I’ll get a TestFlight for it soon. It’s really just missing one more thing before I want to put out the first TestFlight build. Once the TestFlight build is out, I’ll revisit these questions and figure out how I want to build the app going forward.

Here’s what I’m currently considering the main features:

- logging reps with weight
- grouping exercises into a list
- viewing remaining rest time
- seeing exercises completed each day

**Comment by heyjaywilson** [comment]
> having some docs could help to not making bad assumptions

This is 100% spot on though, and I need to be better about documenting these questions. I have a slight vision in mind, but its open to be adjusted or directed in a different way based on feedback after the testflight and some of the already listed issues are taken care of.

I also need to document some developer type things like the background actor thing to write to swiftdata.

**Comment by samuelsainz** [comment]
Appreciate the quick response Jay! Sounds reasonable to me! 

If at some point I can help with anything of this, please let me know, I'd be more than happy to help!
